domain/domain.go

```diff
@@ -66,6 +66,8 @@ import (
 // Domain represents a storage space. Different domains can use the same database name.
 // Multiple domains can be used in parallel without synchronization.
 type Domain struct {
+	context.Context
```

Comment:
https://golang.org/pkg/context/#:~:text=Do%20not%20store%20Contexts%20inside%20a%20struct%20type

The doc suggests "Do not store Contexts inside a struct type".

Reply:
WIth the current code base of resource management, it is the only way without too much modifications.

`Init` and `NewDomain` both needs context. Of course, I should pass context in their callers. But that will involve lots of modifications. So instead of following the idiom mindlessly, I would choose to store it and only modify the domain package itself.

It is a PR that can provide one single context for refactoring other components under domain. Once our code base becomes clean and nice again, it can be removed from the struct.

Reply:
I've added a warning or tip in the latest commit.